code/game/machinery/doors/door.dm

```diff
@@ -459,7 +459,10 @@
 	if(!can_open(forced))
 		return
 	operating = TRUE
-	activate_mobs_in_range(src, 10)
+	for(var/mob/living/G in GLOB.player_list)
```

Comment:
This does not function as discribed
It will run the "G.mind.current" check for every player regardless of location
I.e lets say john guild is in solars and a roach opens a maints door
This check passes and wakes up everything within 15 range

+ lets say pop is 8 this then wakes up everything 8 times in a 15 range.

Reply:
> This does not function as discribed It will run the "G.mind.current" check for every player regardless of location I.e lets say john guild is in solars and a roach opens a maints door This check passes and wakes up everything within 15 range
> 
> * lets say pop is 8 this then wakes up everything 8 times in a 15 range.

During testing with the use of the debug code, this checks for every player that is within the GLOB.player_list

Player list specifically states "//List of all mobs **with clients attached**. Excludes /mob/new_player"

If a roach opens a door and is not in this list, it wont trigger. I've used the debug line to test this as this calls for which mob opened a door. However if it does actually have issues in regards to mobs somehow triggering this, I'll need some more proof because I had breakpointed that line of code and checked to see if actually runs the code, typically there are mobs everywhere on the map that opens doors and my breakpoints only triggered on mobs. 

Could you provide a video with this issue if this is the case however? It should ONLY run on living creatures that have an attached client within the player_list

Unless, what you're saying, is that it triggers the `activate_mob_in_range(src, 15)` with ***every*** client that is currently in the game? If that's the case then woops. Misunderstood. I didn't try testing locally with multiple clients, but I don't think it should trigger twice?

I'd need to test it with a controlled setup with 2 mobs in 2 locations that are far enough apart to not be triggers yet still close enough for both clients to be in range for it to activate I guess. 

I'll look into this later tonight, thank you however for the thought. I had tried to ask how to actually write this code but I asked at a bad time; This should only trigger the line of code once per person however.

Im uh... repeating myself. Anywho.

Reply:
I see; Sorry!